src-tool/Pact/Analyze/Feature.hs

```diff
@@ -1222,6 +1228,63 @@ doc FStringToInteger = Doc
         (TyCon int)
   ]
 
+-- Hash features
+
+doc FStringHash = Doc
+  "hash"
+  CString
+  PropOnly
+  "BLAKE2b 256-bit hash of string values"
+  [ Usage
+      "(hash s)"
+      Map.empty
+      $ Fun
+        Nothing
+        [ ("s", TyCon str)]
+        (TyCon str)
+  ]
+doc FNumericalHash = Doc
+  "hash"
+  CString
+  PropOnly
+  "BLAKE2b 256-bit hash of numerical values"
+  [ let a = TyVar $ TypeVar "a"
+    in Usage
+      "(hash s)"
+      (Map.fromList [("a", OneOf [int, dec])])
+      $ Fun
+        Nothing
+        [ ("s", a)]
+        (TyCon str)
+  ]
+doc FBoolHash = Doc
+  "hash"
+  CLogical
+  PropOnly
+  "BLAKE2b 256-bit hash of bool values"
+  [ Usage
+      "(hash s)"
+      Map.empty
+      $ Fun
+        Nothing
+        [ ("s", TyCon bool)]
+        (TyCon str)
+  ]
+doc FListHash = Doc
+  "hash"
+  CList
+  PropOnly
+  "BLAKE2b 256-bit hash of lists"
```

Comment:
Can we not hash list of lists of primitives, etc.?

Reply:
heterogeneous list can be used in pact 
```
pact> (hash [1 "a"])
"7bjptroWlE_wzrjEZ7QVxzLwVusSnf239sAzbqP9djg"
```
but will fail typechecking with ` Cannot unify: ([integer], [string])`.

But List of Lists .... of the same type should work, good catch